### PR TITLE
bug: sentry fixes

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- https://sentry.io/organizations/gitcoin/issues/1727368446/ (2.6k)
- [bug: 'AssertionError' object has no attribute 'data (4.4k)](https://sentry.io/organizations/gitcoin/issues/2522450092/)
- User.profile.RelatedObjectDoesNotExist
  - https://sentry.io/organizations/gitcoin/issues/1902956247/? (1.1k)
  - https://sentry.io/organizations/gitcoin/issues/1246636419/ 1.3k
  - https://sentry.io/organizations/gitcoin/issues/2170202852 1.1k 
  - https://sentry.io/organizations/gitcoin/issues/2074050238/?project=1398424 (28k)
  - https://sentry.io/organizations/gitcoin/issues/1664409267/?project=1398424 (6.3k)
  - https://sentry.io/organizations/gitcoin/issues/2199790129/ (854)
  - https://sentry.io/organizations/gitcoin/issues/2545687412 (619)
  - https://sentry.io/organizations/gitcoin/issues/1747280089 (535)
  - https://sentry.io/organizations/gitcoin/issues/2458564862 (468)
  - https://sentry.io/organizations/gitcoin/issues/2217158031 (343)
  - https://sentry.io/organizations/gitcoin/issues/2444969956/ (208)
  - https://sentry.io/organizations/gitcoin/issues/2505955385/ (133)
  - https://sentry.io/organizations/gitcoin/issues/2545687412 (42)

##### Refers/Fixes

- GITC-278
- GITC-279
- GITC-280

##### Testing

Tested locally

### CODE REVIEW

**Review by chibie — APPROVED** [review]

**Review by zacheryschiller — COMMENTED** [review]
There are some code scan warnings that GitHub flagged. Have these been addressed?

**Comment by thelostone-mc** [comment]
@zacheryschiller if you notice those aren't related to changes made on this PR ! We can open up a follow up PR to fix those up

**Review by zacheryschiller — APPROVED** [review]

File: app/dashboard/views.py [hunk 1]
```diff
@@ -2161,7 +2161,7 @@ def user_card(request, handle):
     except (ProfileNotFoundException, ProfileHiddenException):
         raise Http404
 
-    if request.user.is_authenticated:
+    if request.user.is_authenticated and request.user.profile:
```
Comment (gdixon) [inline app/dashboard/views.py:2164]: Is this a symptom of a different problem? Why do we have authenticated users without `Profiles`?

Comment (gdixon) [reply app/dashboard/views.py:2164]: ** As discussed we will get this in now and open a new ticket to investigate the cause

**Review by gdixon — APPROVED** [review]

**Pull request merged**